src/main/scala/viper/server/VerificationWorker.scala

```diff
@@ -72,13 +60,40 @@ class VerificationWorker(private val reporterActor: ActorRef,
   }
 
   private var backend: ViperBackend = _
+  implicit val executionContext = ExecutionContext.global
+
+  // Implementation of the Reporter interface used by the backend.
+  class ActorReporter(private val actor_ref: ActorRef,
+                      val tag: String) extends viper.silver.reporter.Reporter {
+
+    val name = s"ViperServer_$tag"
+    var current_offer: Future[QueueOfferResult] = null
+
+    /** Sends massage to the attached actor.
+      *
+      * The actor receving this message offer it to a queue. This offering returns a Future, which will eventually
+      * indicate whether or not the offer was successful. This method is blocking, as it waits for the successful
+      * completion of such an offer.
+      * */
+    def report(msg: reporter.Message): Unit = {
+      implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
+      val answer = actor_ref ? ReporterProtocol.ServerReport(msg)
```

Comment:
Are the responses going to be sent by an instance of ```QueueActor```? 
I cannot find where this happens.

Reply:
Indeed, they are. The relevant lines of code are in the QueueActor's reveice() method. In case it receives a ServerReport(msg) it sends back the future containing the status of the queue offering.

    val offer_status = queue.offer(msg)
    sender() ! offer_status